src/backend/commands/matview.c

```diff
@@ -1843,6 +1850,20 @@ get_prestate_rte(RangeTblEntry *rte, MV_TriggerTable *table,
 	raw = (RawStmt*)linitial(raw_parser(str.data));
 	sub = transformStmt(pstate, raw->stmt);
 
+	/* add securityQuals for tuplestores */
+	foreach(lc, sub->rtable)
+	{
+		RangeTblEntry *rte = (RangeTblEntry *)lfirst(lc);
+		/* if contain UNION ALL clauses, rte has a subquey */
```

Comment:
You means:
"If this query contains UNION ALL clauses, RTEs in rtables has a subquery which contains ENR" ? 

I think that it is straight forward to check setOperations is NIL or not. If this is not NIL, RTEs must have a subquery and this can be confirmed by Assert.

Reply:
> You means:
> "If this query contains UNION ALL clauses, RTEs in rtables has a subquery which contains ENR" ?
That's right.

> I think that it is straight forward to check setOperations is NIL or not. If this is not NIL, RTEs must have a subquery and this can be confirmed by Assert.

Thank you for your alternative suggestion! I use setOperations for check.